src/Common/src/CoreLib/System/IO/Win32Marshal.cs

```diff
@@ -60,7 +60,7 @@ internal static Exception GetExceptionForWin32Error(int errorCode, string path =
                 default:
                     return string.IsNullOrEmpty(path)
                         ? new IOException(GetMessage(errorCode), MakeHRFromErrorCode(errorCode))
-                        : new IOException($"{GetMessage(errorCode)} : '{path}'");
+                        : new IOException($"{GetMessage(errorCode)} : '{path}'", MakeHRFromErrorCode(errorCode));
```

Comment:
This could instead be:
```C#
return new IOException(
    string.IsNullOrEmpty(path) ? GetMessage(errorCode) : $"{GetMessage(errorCode)} : '{path}'",
    MakeHRFromErrorCode(errorCode));
```